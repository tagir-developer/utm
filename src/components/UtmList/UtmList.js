import React, { Fragment } from 'react'
import Utm from '../Utm/Utm'

const UtmList = ({ utmFields, checkedSource, changeSourceFieldsValue, translitHandler }) => {

	const targetSource = utmFields.filter(source => source.name === checkedSource) // "Нам вернулся массив с одним единственным нужным нам объектом"
	const checkedSourceFields = targetSource[0].fields

	const utmsList = (classes, device) => (
		checkedSourceFields.map((utmInfo, index) => {
			return (
				<Utm
					key={device + index}
					utmId={device + index}
					utmInfo={utmInfo}
					classes={classes}
					// popoverInfo={utmInfo.popover}
					changeSourceFieldsValue={changeSourceFieldsValue}
					checkedSource={checkedSource}
					device={device}
				/>
			)
		})
	)

	return (
		<Fragment>
			{/* For PC */}
			<div className="d-none d-md-block">
				<div className="row row-cols-2">
					{utmsList('col-6', 'PC')}
				</div>
			</div>
			{/* For mobile and tablets */}
			<div className="d-block d-md-none">
				{utmsList('mt-1', 'Mobile')}
			</div>
			
			<div className="custom-control custom-checkbox cursor-pointer my-4">
				<input
					className="custom-control-input cursor-pointer"
					type="checkbox"
					id="translitCheckbox"
					onChange={translitHandler}
				/>
				<label className="custom-control-label cursor-pointer" htmlFor="translitCheckbox">
					Транслитерация параметров
 				</label>
			</div>
		</Fragment>
	)
}

UtmList.defaultProps = {
	utmFields: [
		{
			name: 'default source name',
			fields: [{id: '1'}, {id: '2'}, {id: '3'}]
		},
		{
			name: 'default source name (second)',
			fields: [{id: '1'}, {id: '2'}, {id: '3'}]
		}
	], 
	checkedSource: 'default source name', 
	changeSourceFieldsValue: () => {}, 
	translitHandler: () => {}
}

export default UtmList