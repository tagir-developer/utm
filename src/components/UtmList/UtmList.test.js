import React from 'react'
import UtmList from './UtmList'

describe('<UtmList />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<UtmList />)
	})

	describe('Component render. Component props testing', () => {

		it('Should UtmList component render with default props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})

		it('Should render custom UtmList "checkedSource"', () => {
			wrapper.setProps({
				utmFields: [
					{
						name: 'test-name-1',
						fields: [{ id: '1' }, { id: '2' }, { id: '3' }]
					},
					{
						name: 'test-name-2',
						fields: [{ id: '4' }, { id: '5' }, { id: '6' }]
					}
				],
				checkedSource: 'test-name-2',
			})

			wrapper.find('Utm').forEach(node => {
				expect(node.props().checkedSource).toBe('test-name-2')
			})
		})

		it('Should "checkedSource===source.name"', () => {
			wrapper.setProps({
				utmFields: [
					{
						name: 'test-name-1',
						fields: [{ id: '1' }, { id: '2' }, { id: '3' }]
					},
					{
						name: 'test-name-2',
						fields: [{ id: '4' }, { id: '5' }, { id: '6' }]
					}
				],
				checkedSource: 'test-name-2',
			})
				const utmInfoValue = wrapper.find('Utm').at(0).props().utmInfo
				expect(utmInfoValue.id).toBe('4')
		})


	})

	describe('UtmList component handlers testing', () => {

		it('Should call "translitHandler"', () => {
			const changeHandler = jest.fn()
			wrapper.setProps({
				translitHandler: changeHandler,
			})
			expect(changeHandler).toHaveBeenCalledTimes(0)
			wrapper.find('input').simulate('change')
			expect(changeHandler).toHaveBeenCalledTimes(1)
		})

		it('Should use default handler "changeSourceFieldsValue"', () => {
			const result = UtmList.defaultProps.changeSourceFieldsValue()
			expect(result).toBe(undefined)
		})

	})


})