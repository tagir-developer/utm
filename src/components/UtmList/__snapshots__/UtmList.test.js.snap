// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UtmList /> Component render. Component props testing Should UtmList component render with default props 1`] = `
<Fragment>
  <div
    className="d-none d-md-block"
  >
    <div
      className="row row-cols-2"
    >
      <Utm
        changeSourceFieldsValue={[Function]}
        checkedSource="default source name"
        classes="col-6"
        device="PC"
        key="PC0"
        utmId="PC0"
        utmInfo={
          Object {
            "id": "1",
          }
        }
      />
      <Utm
        changeSourceFieldsValue={[Function]}
        checkedSource="default source name"
        classes="col-6"
        device="PC"
        key="PC1"
        utmId="PC1"
        utmInfo={
          Object {
            "id": "2",
          }
        }
      />
      <Utm
        changeSourceFieldsValue={[Function]}
        checkedSource="default source name"
        classes="col-6"
        device="PC"
        key="PC2"
        utmId="PC2"
        utmInfo={
          Object {
            "id": "3",
          }
        }
      />
    </div>
  </div>
  <div
    className="d-block d-md-none"
  >
    <Utm
      changeSourceFieldsValue={[Function]}
      checkedSource="default source name"
      classes="mt-1"
      device="Mobile"
      key="Mobile0"
      utmId="Mobile0"
      utmInfo={
        Object {
          "id": "1",
        }
      }
    />
    <Utm
      changeSourceFieldsValue={[Function]}
      checkedSource="default source name"
      classes="mt-1"
      device="Mobile"
      key="Mobile1"
      utmId="Mobile1"
      utmInfo={
        Object {
          "id": "2",
        }
      }
    />
    <Utm
      changeSourceFieldsValue={[Function]}
      checkedSource="default source name"
      classes="mt-1"
      device="Mobile"
      key="Mobile2"
      utmId="Mobile2"
      utmInfo={
        Object {
          "id": "3",
        }
      }
    />
  </div>
  <div
    className="custom-control custom-checkbox cursor-pointer my-4"
  >
    <input
      className="custom-control-input cursor-pointer"
      id="translitCheckbox"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="custom-control-label cursor-pointer"
      htmlFor="translitCheckbox"
    >
      Транслитерация параметров
    </label>
  </div>
</Fragment>
`;
